api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+/**
+ * @swagger
+ * tags:
+ *   name: Ranks
+ *   description: Rank Management
+ */
+
+import express, { Request, Response } from "express";
+import logger from "../utilities/logger";
+import pool from "../database/index";
+import { Rank } from "../utilities/types";
+import { indexOf } from "lodash";
+import { paginationIsValid } from "../utilities/validation";
+
+const router = express.Router();
+
+const getGameRanks = async (id: number) => {
+  try {
+    const result = await pool.query("SELECT * FROM games WHERE game_id = $1", [
+      id,
+    ]);
+    return result.rows[0].rank_types;
+  } catch (error) {
+    logger.error(error);
+  }
+};
+
+const getAverageRank = async (ranks: Rank[]) => {
+  let average = 0;
+  if (ranks[0].numeric_rank) {
+    ranks.forEach((rank) => (average += rank.numeric_rank));
+    return average / ranks.length;
+  } else {
+    const gameRanks = await getGameRanks(ranks[0].game_id);
+    ranks.forEach((rank) => {
+      average += indexOf(gameRanks, rank.text_rank);
```

Comment:
Should it not be
```suggestion
      average += (indexOf(gameRanks, rank.text_rank) + 1);
```
Or the lowest rank will never be counted towards the average (since it'd do +0)

Reply:
I don't think so? Say the rank indexes are [0,0,1,2,3,4], the average index is 2 so use the second index? Is this not correct?

Reply:
TO-DO; Discuss in person

Reply:
Discussed in person: logic is fine but need to refine the `if` to use a utils function to check if `isNumber`